javascript/modules/client/src/client.ts

```diff
@@ -682,52 +660,31 @@ export class Client extends ClientCore implements IClient {
     try {
       await this.graphql.ensureOnline();
       const client = this.graphql.getClient();
-      const { daos } = await client.request(QueryDaos, {
-        limit,
-        skip,
-        direction,
-        sortBy,
-      });
+      const { daos }: { daos: ISubgraphDaoListItem[] } = await client.request(
+        QueryDaos,
+        {
+          limit,
+          skip,
+          direction,
+          sortBy,
+        },
+      );
       await this.ipfs.ensureOnline();
-      return Promise.all(daos.map(async (dao: any): Promise<DaoListItem> => {
-        // const stringMetadata = await this.ipfs.fetchString(dao.metadata);
-        // TODO
-        // this is a temporal fix and should be changed by the line above
-        // but the current daos in subgraph dont have a proper metadata
-        const stringMetadata = await this.ipfs.fetchString(
-          "QmebY8BGzWBUyVqZFMaFkFmz3JrfaDoFP5orDqzJ1uiEkr",
-        );
-        const metadata = JSON.parse(stringMetadata);
-        return {
-          address: dao.id,
-          ensDomain: dao.name,
-          metadata: {
-            name: metadata.name,
-            avatar: metadata.avatar || null,
-          },
-          // TODO update when new subgraph schema is deployed
-          plugins: dao.packages.map(
-            (
-              plugin: SubgraphPlugin,
-            ): InstalledPluginListItem => {
-              return {
-                instanceAddress: plugin.pkg.id,
-                // TODO
-                // this should be the address of the parent plugin
-                // or the ens name, subgraph does not have this info
-                // at this moment so for now, the __typename
-                // will do the job to differentitate between
-                // plugins, however, this will eventually change
-                id: plugin.pkg.__typename + "plugin.dao.eth",
-                version: "0.0.1",
-              };
-            },
-          ),
-        };
-      }));
-      // return daosDetails;
+      return Promise.all(
+        daos.map(async (dao: ISubgraphDaoListItem): Promise<DaoListItem> => {
+          // const stringMetadata = await this.ipfs.fetchString(dao.metadata);
+          // TODO
+          // this is a temporal fix and should be changed by the line above
+          // but the current daos in subgraph dont have a proper metadata
+          const stringMetadata = await this.ipfs.fetchString(
+            "QmebY8BGzWBUyVqZFMaFkFmz3JrfaDoFP5orDqzJ1uiEkr",
+          );
+          const metadata = JSON.parse(stringMetadata);
+          return toDaoListItem(dao, metadata);
+        }),
```

Comment:
Turning this into raw promises would be better for the final bundle size. Same on the rest of Promise.all() where there's no `if/else`


```suggestion
        daos.map((dao: ISubgraphDaoListItem): Promise<DaoListItem> => {
          // const stringMetadata = await this.ipfs.fetchString(dao.metadata);
          // TODO
          // this is a temporal fix and should be changed by the line above
          // but the current daos in subgraph dont have a proper metadata

          return this.ipfs.fetchString(
            "QmebY8BGzWBUyVqZFMaFkFmz3JrfaDoFP5orDqzJ1uiEkr",
          ).then(stringMetadata => {
            const metadata = JSON.parse(stringMetadata);
            return toDaoListItem(dao, metadata);
          });
        }),
```